### PR TITLE
feat: `Id()` now accepts unnamed components

### PR DESCRIPTION
**Pull request by hadley**
I think requiring names has little benefit — we never actually use them, never need to care about them, and requires a bunch of extra typing.

### CODE REVIEW

**Review by krlmlr — COMMENTED** [review]
Thanks, good idea.

File: R/dbQuoteIdentifier_DBIConnection.R [hunk 1]
```diff
@@ -37,4 +34,4 @@ setMethod("dbQuoteIdentifier", signature("DBIConnection", "SQL"), dbQuoteIdentif
 
 #' @rdname hidden_aliases
 #' @export
-setMethod("dbQuoteIdentifier", signature("DBIConnection", "Id"), dbQuoteIdentifier_DBIConnection)
+setMethod("dbQuoteIdentifier", signature("DBIConnection", "Id"), dbQuoteIdentifier_DBIConnection_Id)
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/dbQuoteIdentifier_DBIConnection.R:37]: This doesn’t change the dispatch, just the implementation

[the remark this answers is not in the document]
Comment (krlmlr) [reply R/dbQuoteIdentifier_DBIConnection.R:37]: Good catch. It's been like this for 4+ years now, there's really no reason to keep the switchpatch. I might be able to disentangle the other cases too.

**Comment by krlmlr** [comment]
Thanks!

**Comment by hadley** [comment]
I assume this somehow gets turn into an automatic news bullet?

**Pull request merged**